service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/EhrExtractStatusService.java

```diff
@@ -161,6 +175,38 @@ public EhrExtractStatus updateEhrExtractStatusContinue(String conversationId) {
         return ehrExtractStatus;
     }
 
+    public EhrExtractStatus updateEhrExtractStatusAck(String conversationId, EhrReceivedAcknowledgement ack) {
+        Query query = createQueryForConversationId(conversationId);
+
+        Update update = createUpdateWithUpdatedAt();
+        update.set(RECEIVED_ACK_TIMESTAMP, ack.getReceived());
+        update.set(RECEIVED_ACK_CONVERSATION_CLOSED, ack.getConversationClosed());
+        update.set(RECEIVED_ACK_ROOT_ID, ack.getRootId());
+        update.set(RECEIVED_ACK_MESSAGE_REF, ack.getMessageRef());
+
+        if (!isEmpty(ack.getErrors())) {
+            ack.getErrors().stream()
+                .forEach(error -> update.addToSet(RECEIVED_ACK_ERRORS, ErrorDetails.builder()
```

Comment:
`.getErrors()` returns a `List` which has its own `.forEach()` - so no need to go via `.stream()` manually.

Reply:
And elsewhere